workflow/scripts/collapse_lanes.py

```diff
@@ -0,0 +1,127 @@
+import shutil
+import os
+import logging
+from datetime import datetime
+from itertools import chain
+from pathlib import Path
+
+
+logging.basicConfig(level=logging.INFO)
+sampleid = lambda x: str(x.stem).split('_')[0]
+repid = lambda x: str(x.stem).split('_')[2]
+tmp_dir = Path('/', 'gpfs', 'gsfs12', 'users', 'NHLBI_IDSS', 'rawdata', 'metamorph-test', 'merged', 'rna', 'tmp')
+
+
+def gunzip(from_gzfile, to_regfile):
+    """
+        Uncompress a file from a gzip compressed file to a regular uncompressed file
+
+        @param from_gzfile <pathlib.Path> or <str>, gzipped from file
+        @param to_regfile <pathlib.Path> or <str>, regular file output
+    """
+    with gzip.open(from_gzfile, 'rb') as f_in:
+        with open(to_regfile, 'wb') as f_out:
+            shutil.copyfileobj(f_in, f_out)
+    return Path(to_regfile).exists()
+
+
+def catfiles(outputfile, *files):
+    """
+        Concatenate multiple files into one
+
+        @param outputfile <pathlib.Path> or <str>, file path to ouput concatination to
+        @param *files <list>, list of valid file paths to concatinate
+    """
+    with open(outputfile, 'wb') as wfd:
+        for f in files:
+            with open(f,'rb') as fd:
+                shutil.copyfileobj(fd, wfd)
+    return Path(outputfile).exists()
+
+
+def main(args):
+    """
+        Collapse multiplex samples across lanes into single fastq.gz files. Optimally uncompressing first and recompressing.
+        This script matches partner fastq.gz's by the a common sample identifier and lane identifier. Then uses lexicographical
+        order to merge each consituent file into one.
+
+        E.g.:
+            Control.rna62_L001_R1.fastq.gz and Control.rna62_L004_R1.fastq.gz
+            Control.rna62_L001_R2.fastq.gz and Control.rna62_L004_R2.fastq.gz
+        File name template:
+            <sample id>_<lane id>_<replicate id>.fastq.gz
+
+        @param args.outputdir <pathlib.Path> output directory
+        @param args.inputdir <pathlib.Path> input directory
+    """
+
+    args.inputdir = Path(args.inputdir).resolve()
+    args.outputdir = Path(args.outputdir).resolve()
+    
+    if not args.outputdir.exists():
+        Path.mkdir(args.outputdir)
+
+    # pairing
+    all_files = sorted(args.inputdir.glob('*.fastq.gz'))
+    pairs = []
+    for _file in all_files:
+        if _file in list(chain.from_iterable(pairs)):
+            continue
+        sid = sampleid(_file)
+        rid = repid(_file)
+        partner = None
+        for _partner_file in all_files:
+            if _file == _partner_file: continue
+            partner_sid = sampleid(_partner_file)
+            partner_rid = repid(_partner_file)
+            if partner_sid == sid and partner_rid == rid:
+                if partner is not None:
+                    logging.warn(
+                        f'More than two files with the sample identifier: {partner_sid}\n'
+                        f'--- Files: {_file}, {_partner_file}, {partner}'
```

Comment:
This also needs to be addressed once this script is fully functional. Read cans be disrupted over >2 lanes.